builtin/joined_dataset.h

```diff
@@ -25,6 +25,7 @@ struct JoinedDatasetConfig {
     std::shared_ptr<TableExpression> left;
     std::shared_ptr<TableExpression> right;
     std::shared_ptr<SqlExpression> on;
+    JoinQualification qualification;
```

Comment:
We should consider grouping these fields into one SQL expression as we have done at many places.